types/denom_test.go

```diff
@@ -0,0 +1,77 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	atom  = "atom"  // 1 (base denom unit)
+	matom = "matom" // 10^-3 (milli)
+	uatom = "uatom" // 10^-6 (micro)
+	natom = "natom" // 10^-9 (nano)
+)
+
+func TestRegisterDenom(t *testing.T) {
+	atomUnit := OneDec() // 1 (base denom unit)
+
+	require.NoError(t, RegisterDenom(atom, atomUnit))
+	require.Error(t, RegisterDenom(atom, atomUnit))
+
+	res, ok := GetDenomUnit(atom)
+	require.True(t, ok)
+	require.Equal(t, atomUnit, res)
+
+	res, ok = GetDenomUnit(matom)
+	require.False(t, ok)
+	require.Equal(t, ZeroDec(), res)
+}
+
+func TestConvertCoins(t *testing.T) {
+	atomUnit := OneDec() // 1 (base denom unit)
+	require.NoError(t, RegisterDenom(atom, atomUnit))
+
+	matomUnit := NewDecWithPrec(10, 4) // 10^-3 (milli)
```

Comment:
```suggestion
	matomUnit := NewDecWithPrec(1, 3) // 10^-3 (milli)
```

OR `matomUnit := MustNewDecFromStr("0.001")`